oauthenticator/oauth2.py

```diff
@@ -680,13 +680,13 @@ def _allowed_scopes_validation(self, proposal):
         """,
     )
 
-    pkce = Bool(
+    require_pkce = Bool(
         False,
         config=True,
         help="""
-        Whether to use PKCE (Proof Key for Code Exchange) for the OAuth2 flow.
-
-        Only the S256 method is supported.
+        Require Proof Key for Code Exchange (PKCE) for OAuth2 authorization code flow
```

Comment:
Is this to say that the oauth server requires pkce, so that the client is configured to say the server requires it? Or, is it that the client is configured to require it independent of the server, or for servers that supports it optionally and/or requires it?

I'm not clear about the situation, but i hope that these confusions can be clarified in the docstring.

Reply:
Thank you @renan-r-santos for working this!!!

Reply:
I think this is what confused me at first- based on the discussions in this PR only the server can require PKCE, but a client can't.